traits/trait_types.py

```diff
@@ -1386,13 +1313,43 @@ class PythonValue(Any):
     metadata = {"editor": shell_editor}
 
 
-# -------------------------------------------------------------------------------
-#  'BaseFile' and 'File' traits:
-# -------------------------------------------------------------------------------
-
-
 class BaseFile(BaseStr):
     """ Defines a trait whose value must be the name of a file.
+
+    For Python 3.6 this will accept os.pathlib Path objects, converting them
```

Comment:
Suggest adding "and later", or using "Python >= 3.6". (Also applies to the `File` docstring.)